Commonalities/documentation/API-design-guidelines.md

```diff
@@ -1021,3 +1027,245 @@ Finally, this part describes the OAuth security applied to the API. This spec is
+
+Providing this capability is optional for any CAMARA APIs depending on UC requirements
+
+If this capability is present in CAMARA API, following attributes **must** be used in the POST request for the managed entity:
+
+| attribute name | type | attribute description | cardinality |
+| ----- |	-----  |	 -----  | -----  | 
+| notificationUrl | string | https callback address where the notification must be POST-ed | mandatory |
+| notificationAuthToken | string | authentication token for callback API endpoint. It MUST be indicated within HTTP Authorization header as a Bearer Token format, e.g. ```Authorization: Bearer $notificationAuthToken``` | optional |
+
+Format conventions regarding ```notificationAuthToken``` attribute, in order to provide Uniqueness, Randomness and Simplicity for its management are the following:
+- It MUST BE an opaque attribute
+- It MUST NOT be a JWT Token
+- It has to be a restricted length, being an string between [20-64] characters.
+- It is HIGHLY recommended to have random-based pattern, like UUIDv4 format (32 hexadecimal characters, separated by `-`, e.g. ```550e8400-e29b-41d4-a716-446655440000```) 
+
+**Resource-based subscription**
+
+A resource-based subscription is a subscription managed as a resource. An endpoint is provided to request subscription creation.  As this subscription is managed as an API resource, it is identified and operations to search, retrieve and delete them must be provided.
+
+Note: It is perfectly valid for a CAMARA API to have several event types managed. The subscription endpoint will be unique but 'eventType' attribute is used to distinguish distinct events subscribed.
+
+In order to ease developer adoption, the pattern for Resource-based subscription **must** be consistent for all API providing this feature.
+
+4 operations must be defined:
+
+| operation | path | description |
+| ----- |	-----  |	 -----  | 
+| POST | `/subscriptions` | Operation to request a subscription.     |
+| GET | `/subscriptions` | Operation to retrieve a list of subscriptions - could be an empty list.  eg. `GET /subscriptions?type=ROAMING_STATUS&ExpireTime.lt=2023-03-17` |
+| GET | `/subscriptions/{subscriptionId}` | Operation to retrieve a subscription |
+| DELETE | `/subscriptions/{subscriptionId}` | Operation to delete a subscription |
+
+
+Following table provides subscriptions attributes
+
+| name | type | attribute description | cardinality |
+| ----- |	-----  |	 -----  |  -----  | 
+| notificationUrl | string | https Callback address where the notification must be POST-ed | mandatory |
+| notificationAuthToken | string | Authentification token for callback API | optional |
```

Comment:
See my [comment](https://github.com/camaraproject/WorkingGroups/pull/173#discussion_r1180347944) above. It doesn't matter to the API provider (sending notifications) how the token is used when received by the client. They just need to send it in an `Authorization: Bearer` header so that, **if it is** an OAuth token, then it will do its job. But it does not **need** to be an OAuth token.

Reply:
I have updated my concerns in the earlier comment.

Besides, line 1143 shows this `token` used in `data` instead of `Authorization` header.

Reply:
I've updated the example in order to show the token in the `POST notifications`; 

There are also rational why  `notificationAuthToken`  is in the the body (data) of the POST subscriptions (thanks to @eric-murray )

@sfnuser may I consider this comment resolved?

Reply:
I shall close this comment on the aspect that `notificationAuthToken` being in the body.